packages/shared/test/shared-secret-compatibility.test.ts

```diff
@@ -0,0 +1,271 @@
+ *
+ * Test vectors are loaded from a JSON file which makes it easier to:
+ * 1. Share vectors between different implementations (Rust, TypeScript)
+ * 2. Update or add vectors without changing test logic
+ * 3. Maintain expected inputs/outputs separately from test code
+ */
+describe('Shared Secret Compatibility Tests', () => {
+  // Parse test vectors from the loaded JSON file
+  const { test_vectors } = testVectorsFile as { test_vectors: any[] };
+
+  // Helper to convert test vector from JSON to usable objects
+  const prepareTestVector = (vector: any) => {
+    return {
+      id: vector.id,
+      description: vector.description,
+      sharedSecret: new Uint8Array(vector.shared_secret),
+      plaintext: vector.plaintext
+        ? new TextEncoder().encode(vector.plaintext)
+        : new Uint8Array(0),
+      fixedNonce: new Uint8Array(vector.fixed_nonce),
+      expectedCiphertext: vector.expected_ciphertext
+        ? Buffer.from(vector.expected_ciphertext, 'hex')
+        : null,
+      rustGeneratedCiphertext: vector.rust_generated_ciphertext
+        ? new Uint8Array(vector.rust_generated_ciphertext)
+        : null,
+      expectedPlaintext: vector.expected_plaintext,
+    };
+  };
+
+  // Parse test vectors into usable formats
+  const testVector1 = prepareTestVector(test_vectors[0]);
+  const testVector2 = prepareTestVector(test_vectors[1]);
+  const rustCompatVector = prepareTestVector(test_vectors[2]);
+
+  // Set up a fixed nonce for testing
+  function setupFixedNonceMock(fixedNonce: Uint8Array) {
+    // Get the mocked function
+    const mockedRandomBytes = vi.mocked(randomBytes);
+
+    // Clear previous implementations
+    mockedRandomBytes.mockReset();
+
+    // Configure it to return our fixed nonce
+    mockedRandomBytes.mockImplementation((bytesLength?: number) => {
+      // If length is undefined or matches our nonce length, return the fixed nonce
+      // Otherwise return a new array of the requested length
+      if (bytesLength === undefined || bytesLength === fixedNonce.length) {
+        return fixedNonce;
+      } else {
+        return new Uint8Array(bytesLength);
+      }
+    });
+
+    // Return a cleanup function
+    return () => {
+      mockedRandomBytes.mockReset();
+    };
+  }
+
+  describe('Round-trip fixed vector tests', () => {
+    // Testing each vector for round-trip compatibility
+    it('should correctly encrypt and decrypt test vector 1', () => {
+      const { sharedSecret, plaintext, fixedNonce, expectedCiphertext } =
+        testVector1;
+
+      // Set up our mock to return the fixed nonce
+      const cleanupMock = setupFixedNonceMock(fixedNonce);
+
+      try {
+        // Encrypt with fixed nonce
+        const ciphertext = encryptWithSharedSecret(sharedSecret, plaintext);
+
+        // Check against expected ciphertext from the JSON file
+        const ciphertextHex = Buffer.from(ciphertext).toString('hex');
+        if (expectedCiphertext) {
```

Comment:
Is there a case where this wouldn't be set for the test vectors i.e. do we need this if-else block? Would the test simply fail spectacularly if `Buffer.from(expectedCiphertext)` (L102) is called on an undefined object, and that would be sufficient?